File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.java
package com.getstream.sdk.chat.adapter;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.getstream.sdk.chat.adapter.viewholder.attachment.BaseAttachmentViewHolder;
import com.getstream.sdk.chat.view.MessageListViewStyle;

import java.util.List;

import kotlin.collections.CollectionsKt;


public class AttachmentListItemAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private final AttachmentViewHolderFactory factory;
    private final MessageListViewStyle style;

    private final MessageListItem.MessageItem messageListItem;
    private final List<AttachmentListItem> attachments;

    public AttachmentListItemAdapter(@NonNull MessageListItem.MessageItem messageListItem,
                                     @NonNull AttachmentViewHolderFactory factory,
                                     @NonNull MessageListViewStyle style
    ) {
        this.factory = factory;
        this.messageListItem = messageListItem;
        this.attachments = CollectionsKt.map(messageListItem.getMessage().getAttachments(), AttachmentListItem::new);
        this.style = style;
    }

    @Override
    public int getItemViewType(int position) {
        return factory.getAttachmentViewType(attachments.get(position));
    }

    @Override
    public int getItemCount() {
        return attachments.size();
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent,
                                                      int viewType) {
        return this.factory.createAttachmentViewHolder(parent, viewType, style, messageListItem);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        AttachmentListItem attachmentItem = attachments.get(position);
        ((BaseAttachmentViewHolder) holder).bind(attachmentItem);
    }

}

File: library/src/main/java/com/getstream/sdk/chat/adapter/AttachmentListItemAdapter.kt
package com.getstream.sdk.chat.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.getstream.sdk.chat.adapter.viewholder.attachment.BaseAttachmentViewHolder
import com.getstream.sdk.chat.view.MessageListViewStyle
import io.getstream.chat.android.livedata.utils.MessageListItem

class AttachmentListItemAdapter(
        private val messageListItem: MessageListItem.MessageItem,
        private val factory: AttachmentViewHolderFactory,
        private val style: MessageListViewStyle
) : RecyclerView.Adapter<BaseAttachmentViewHolder>() {

    private val attachments: List<AttachmentListItem> =
        messageListItem.message.attachments.map(::AttachmentListItem)

    override fun getItemViewType(position: Int): Int {
        return factory.getAttachmentViewType(attachments[position])
    }

    override fun getItemCount() = attachments.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseAttachmentViewHolder {
        return factory.createAttachmentViewHolder(parent, viewType, style, messageListItem)
    }

    override fun onBindViewHolder(holder: BaseAttachmentViewHolder, position: Int) {
        holder.bind(attachments[position])
    }
}
